packages/react-scripts/scripts/utils/createJestConfig.js

```diff
@@ -40,7 +40,7 @@ module.exports = (resolve, rootDir, isEjecting) => {
     setupTestFrameworkScriptFile: setupTestsFile,
     testMatch: [
       '<rootDir>/src/**/__tests__/**/*.{js,jsx,ts,tsx}',
-      '<rootDir>/src/**/?(*.)(spec|test).{js,jsx,ts,tsx}',
+      '<rootDir>/src/**/*(*.){spec,test}.{js,jsx,ts,tsx}',
```

Comment:
I'm not a micromatch expert, but should this be `<rootDir>/src/**/*.{spec,test}.{js,jsx,ts,tsx}` instead? Not sure on the difference in matching between the two.

Reply:
`*(*.)` is needed to detect /spec.js and /test.js files.

Reply:
Maybe i misunderstood, but we don't want to match `test.js` and `spec.js` specifically. We only want to match files with `.test.js`/`.spec.js` suffix, or files within a `__tests__` directory

Reply:
For reference here's the Jest 24 default `testMatch`: https://github.com/facebook/jest/blob/8c128faca3b9b39bf888a01c98cdef8b3a373b0b/packages/jest-config/src/Defaults.ts#L70

Would `**/?(*.)+(spec|test).[tj]s?(x)` solve our issues?

Reply:
No, this is not working on windows. 
If you don't want to match test.js and spec.js, then `<rootDir>/src/**/*.{spec,test}.{js,jsx,ts,tsx}` should be used.
Otherwise `<rootDir>/src/**/*(*.){spec,test}.{js,jsx,ts,tsx}` works fine.

Reply:
Matching `test.js` and `spec.js` was previous behavior. If that is going to change, then it is a breaking change that will need to be highlighted in the release notes.

Reply:
We've never claimed to support matching `test.js` and `spec.js` outside of `__tests__` [in our documentation](https://facebook.github.io/create-react-app/docs/running-tests#filename-conventions). I think this is a safe change to make for v3 and lines up with how we've always communicated the filename conventions in our docs.